modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc

```diff
@@ -78,5 +97,70 @@ Eigen::Matrix<float, 3, 1> FlatCameraTransformer::MakeUnit(
   return unit_v;
 }
 
+bool FlatCameraTransformer::HaveHighConfidence(
+  std::shared_ptr<VisualObject> obj_ptr) {
+    if (obj_ptr->trunc_width > 0.25f) return false;
+    if (obj_ptr->trunc_height > 0.25f) return false;
+    if (obj_ptr->distance > 50.0) return false;
+    if (obj_ptr->distance < 5.0) return false;
+
+    double azimuth = std::atan2(obj_ptr->center[1], obj_ptr->center[0])
+     * 180.0 / M_PI;
+    if (!(-30.0 < azimuth && azimuth < 30.0)) return false;
+
+    return true;
+}
+
+void FlatCameraTransformer::GetDynamicExtrinsics(
+  const std::vector<std::pair<Eigen::Vector3f, Eigen::Vector3f>> &centers) {
+    if (centers.empty()) {
+      adjust_pitch_ = false;
+      camera2car_adj_ = camera2car_;
+      pitch_diff_ = 0.0f;
+      return;
+    }
+
+    float min_diff = std::numeric_limits<float>::max();
+    float best_pitch_adjustment = 0.0f;
+    Eigen::Matrix<float, 4, 4> best_camera2car = camera2car_;
+
+    Eigen::Matrix<float, 4, 4> rot;
+    rot.setIdentity();
+    for (float p = -3.0; p < 3.0; p += 0.2f) {
```

Comment:
@eugenegx Are you adjusting along the yawl angle (of the vehicle) instead? between -3.0 and 3.0 with 0.2 degree of increment?

Reply:
This is pitch :)

Reply:
Sorry my bad, camera2car_ already converted into FLU